### PR TITLE
Automated Google Search on Firefox

### PR DESCRIPTION
**Pull request by Antonieta13**
Hola Buenos dias 

Esto esta con firefox, quiero probar tambien con chrome esta bien?

### CODE REVIEW

File: src/test/java/GoogleSearch.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        driver.findElement(By.name("q")).sendKeys("Testing");
+
+        sleep(1000);
+        driver.quit();
+
+    }
+
+    private void sendKeys(String testing) {
+    }
```
Comment (yfarfan) [inline src/test/java/GoogleSearch.java]: el metodo sendKeys esta vacio.

File: src/test/java/GoogleSearch.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import static java.lang.Thread.sleep;
+
+public class GoogleSearch {
+
+    @Test
+    public void webdriverFirefox() throws InterruptedException {
+        System.setProperty("webdriver.gecko.driver", "C:/geckoDriver/geckodriver.exe");
+
```
Comment (yfarfan) [inline src/test/java/GoogleSearch.java:13]: Puedes crear el driver como un atributo de la clase  GoogleSearch

File: src/test/java/GoogleSearch.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        WebDriver driver = new FirefoxDriver();
+
```
Comment (yfarfan) [inline src/test/java/GoogleSearch.java:23]: Puedes inicializar esto en un metodo que tenga la anotacion @Before para que se ejecute antes del test

**Review by yfarfan — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 4]
```diff
@@ -12,6 +12,9 @@ repositories {
 dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.0'
+    // https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java
+    implementation group: 'org.seleniumhq.selenium', name: 'selenium-java', version: '4.0.0-beta-3'
```
Comment (aperdomob) [inline build.gradle]: Utiliza la versión 3 para evitar comportamientos extraños, que no sea beta

File: src/test/java/GoogleSearch.java (hunk 2 above)
Comment (aperdomob) [inline src/test/java/GoogleSearch.java:22]: @yfarfan que significa esto?

File: src/test/java/GoogleSearch.java (hunk 1 above)
Comment (aperdomob) [reply src/test/java/GoogleSearch.java]: Elimina este metodo

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 4 above)
Comment (Antonieta13) [reply build.gradle]: Listo

File: src/test/java/GoogleSearch.java (hunk 1 above)
Comment (Antonieta13) [reply src/test/java/GoogleSearch.java]: Eliminado

File: src/test/java/GoogleSearch.java (hunk 3 above)
Comment (yfarfan) [reply src/test/java/GoogleSearch.java:23]: Y la variable driver colocarla como un atributo de la clase, y el **new** se hace dentro del @Before de la prueba

File: src/test/java/GoogleSearch.java (hunk 2 above)
Comment (Antonieta13) [reply src/test/java/GoogleSearch.java:22]: Alejandro el Gecko lo necesito para correr firefox, si lo quito me arroja este error. 

No se si esa es la pregunta

The path to the driver executable must be set by the webdriver.gecko.driver system property; for more information, see https://github.com/mozilla/geckodriver. The latest version can be downloaded from https://github.com/mozilla/geckodriver/releases
java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.gecko.driver system property; for more information, see https://github.com/mozilla/geckodriver. The latest version can be downloaded from https://github.com/mozilla/geckodriver/releases
	at com.google.common.base.Preconditions.checkState(Preconditions.java:847)
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:134)
	at org.openqa.selenium.firefox.GeckoDriverService.access$100(GeckoDriverService.java:44)
	at org.openqa.selenium.firefox.GeckoDriverService$Builder.findDefaultExecutable(GeckoDriverService.java:167)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:355)
	at org.openqa.selenium.firefox.FirefoxDriver.toExecutor(FirefoxDriver.java:190)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:147)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:125)
	at GoogleSearch.before(GoogleSearch.java:19)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptLifecycleMethod(TimeoutExtension.java:126)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptBeforeAllMethod(TimeoutExtension.java:68)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllMethods$9(ClassBasedTestDescriptor.java:384)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllMethods(ClassBasedTestDescriptor.java:382)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:196)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:834)

File: src/test/java/GoogleSearch.java (hunk 3 above)
Comment (Antonieta13) [reply src/test/java/GoogleSearch.java:23]: Modificado

File: build.gradle [hunk 5]
```diff
@@ -10,8 +10,13 @@ repositories {
 }
 
 dependencies {
+    implementation 'junit:junit:4.13.1'
```
Comment (aperdomob) [inline build.gradle]: remove

File: src/test/java/GoogleSearch.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        System.out.println("before()");
+        System.setProperty("webdriver.gecko.driver", "C:/geckoDriver/geckodriver.exe");
+        GoogleSearch.Driver = new FirefoxDriver();
+        System.out.println(Driver.hashCode());
+    }
+
+    @Test
+    public void webdriverFirefox() throws InterruptedException {
+
+        System.out.println(Driver);
```
Comment (aperdomob) [inline src/test/java/GoogleSearch.java]: remove

File: src/test/java/GoogleSearch.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        sleep(2000);
+
+        Driver.findElement(By.name("q")).sendKeys("Testing");
+
+        sleep(1000);
+        Driver.quit();
+
+    }
+    
+}
```
Comment (aperdomob) [inline src/test/java/GoogleSearch.java]: agrega al final siempre una linea en limpio, de esa forma no te sale ese icono rojito, y la recomendación es tener esa linea al final

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 5 above)
Comment (Antonieta13) [reply build.gradle]: listo

File: build.gradle [hunk 8]
```diff
@@ -10,8 +10,13 @@ repositories {
 }
 
 dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
+    implementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
+    //testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
```
Comment (aperdomob) [inline build.gradle]: remove these lines

File: src/test/java/GoogleSearch.java (hunk 6 above)
Comment (Antonieta13) [reply src/test/java/GoogleSearch.java]: listo

File: src/test/java/GoogleSearch.java (hunk 7 above)
Comment (Antonieta13) [reply src/test/java/GoogleSearch.java]: listo

File: src/test/java/GoogleSearch.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import static java.lang.Thread.sleep;
+
+public class GoogleSearch {
+
+    private FirefoxDriver Driver;
```
Comment (aperdomob) [inline src/test/java/GoogleSearch.java]: the convention for variables in Java is camelcase, for this case  change to `driver`

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 8 above)
Comment (Antonieta13) [reply build.gradle]: Done

File: src/test/java/GoogleSearch.java (hunk 9 above)
Comment (Antonieta13) [reply src/test/java/GoogleSearch.java]: Done

**Review by aperdomob — APPROVED** [review]

**Review by yfarfan — APPROVED** [review]

**Pull request merged**